pkg/scalers/cpu_memory_scaler.go

```diff
@@ -84,16 +130,38 @@ func parseResourceMetadata(config *scalersconfig.ScalerConfig, logger logr.Logge
 	case v2.AverageValueMetricType:
 		averageValueQuantity := resource.MustParse(meta.Value)
 		meta.AverageValue = &averageValueQuantity
+
+		activationValueQuantity := resource.MustParse(meta.ActivationValue)
+		meta.ActivationAverageValue = &activationValueQuantity
 	case v2.UtilizationMetricType:
 		utilizationNum, err := parseUtilization(meta.Value)
 		if err != nil {
 			return meta, err
 		}
 		meta.AverageUtilization = utilizationNum
+
+		utilizationNum, err = parseUtilization(meta.ActivationValue)
+		if err != nil {
+			return meta, err
+		}
+		meta.ActivationAverageUtilization = utilizationNum
 	default:
 		return meta, fmt.Errorf("unknown metric type: %s, allowed values are 'Utilization' or 'AverageValue'", string(meta.MetricType))
 	}
 
+	if config.ScalableObjectType == "ScaledObject" {
```

Comment:
we should fail for other types imho

Reply:
Are you suggesting that I do not check for `config.ScalableObjectType` in the current code, and instead return an error for other types? If that's so, should I parse the error string to see if the error is related to type? Because `config.ScalableObjectType` not being `ScaledObject` should not result in the error of the `parseResourceMetadata()` function.

Reply:
basically add else and if the type is different fire an error mentioning that the type is not supported